### PR TITLE
[vp] Added a test system for wavevisiond talking to controlsd

### PR DESCRIPTION
**Pull request by railsnoob**

### CODE REVIEW

File: selfdrive/controls/controlsd.py [hunk 1]
```diff
@@ -46,6 +46,8 @@ def controlsd_thread(gctx, rate=100):  #rate in Hz
   model = messaging.sub_sock(context, service_list['model'].port)
   health = messaging.sub_sock(context, service_list['health'].port)
 
+  waveModel = messaging.sub_sock(contexst, service_list['waveModel'].port)
```
Comment (BillZito) [inline selfdrive/controls/controlsd.py]: typo, context* not contexst

Comment (BillZito) [inline selfdrive/controls/controlsd.py]: typo, context* not contexst

File: selfdrive/wavevisiond/wavevisiond.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import zmq
+import selfdrive.messaging as messaging
+from common.services import service_list
+import numpy as np
+from common.realtime import sec_since_boot, set_realtime_priority, Ratekeeper
+
+import time
+
+def wavevisiond_thread(gctx,rate=100):
```
Comment (BillZito) [inline selfdrive/wavevisiond/wavevisiond.py]: why 100?

File: selfdrive/wavevisiond/wavevisiond.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        msg.leftLane.points = [5.0, 4.0, 3.0, 2.0, 1.0 ]
+        msg.leftLane.prob = 0.987
+        msg.leftLane.std = 0.1
+
+        waveModel.send(msg.to_bytes())
+        time.sleep(6)
+    
+    
+def main(gctx=None):
+    wavevisiond_thread(gctx, rate)
```
Comment (BillZito) [inline selfdrive/wavevisiond/wavevisiond.py]: rate needs to be defined as 100 in main definition, not defined here

File: selfdrive/wavevisiond/wavevisiond.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    while True:
+        msg = messaging.new_message()
+        msg.init('model')
+        msg.path.points=[1.0,5.0,4.0,4.0,5.0]
+        msg.path.prob = 0.743
+        msg.path.std = 0.2
+
+        msg.leftLane.points = [5.0, 4.0, 3.0, 2.0, 1.0 ]
+        msg.leftLane.prob = 0.987
+        msg.leftLane.std = 0.1
```
Comment (BillZito) [inline selfdrive/wavevisiond/wavevisiond.py]: no right lane... I guess we don't need that to test.

File: selfdrive/wavevisiond/wavevisiond.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+def wavevisiond_thread(gctx,rate=100):
+    set_realtime_priority(1)
+    
+    context = zmq.Context()
+
+    waveModel = messaging.pub_sock(context, service_list['waveModel'].port)
+    
+    while True:
+        msg = messaging.new_message()
+        msg.init('model')
```
Comment (BillZito) [inline selfdrive/wavevisiond/wavevisiond.py]: this should be init waveModel, right?

File: selfdrive/wavevisiond/wavevisiond.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import zmq
+import selfdrive.messaging as messaging
+from common.services import service_list
+import numpy as np
+from common.realtime import sec_since_boot, set_realtime_priority, Ratekeeper
+
+import time
+
+def wavevisiond_thread(gctx,rate=100):
+    set_realtime_priority(1)
```
Comment (BillZito) [inline selfdrive/wavevisiond/wavevisiond.py]: why is this line necessary?

File: selfdrive/controls/controlsd.py (hunk 1 above)
Comment (railsnoob) [inline selfdrive/controls/controlsd.py]: Yes context. Fixed.

File: selfdrive/wavevisiond/wavevisiond.py (hunk 2 above)
Comment (railsnoob) [inline selfdrive/wavevisiond/wavevisiond.py]: Thats for future.

File: selfdrive/wavevisiond/wavevisiond.py (hunk 6 above)
Comment (railsnoob) [inline selfdrive/wavevisiond/wavevisiond.py]: It sets real time priority. Right now it makes sure it runs at lower priority than controlsd.

File: selfdrive/wavevisiond/wavevisiond.py (hunk 5 above)
Comment (railsnoob) [inline selfdrive/wavevisiond/wavevisiond.py]: No. It should be model. Its got to do with how the message is formatted.

File: selfdrive/wavevisiond/wavevisiond.py (hunk 4 above)
Comment (railsnoob) [inline selfdrive/wavevisiond/wavevisiond.py]: Yep.

File: selfdrive/wavevisiond/wavevisiond.py (hunk 3 above)
Comment (railsnoob) [inline selfdrive/wavevisiond/wavevisiond.py]: yep
